src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -593,17 +593,23 @@ Copyright (c) .NET Foundation. All rights reserved.
            targets import these via the MSBuildProjectExtensionsPath, so that's what the RestoreOutputPath should be. -->
       <RestoreOutputPath Condition=" '$(RestoreOutputPath)' == '' " >$(MSBuildProjectExtensionsPath)</RestoreOutputPath>
     </PropertyGroup>
+
+    <ConvertToAbsolutePath Paths="$(RestoreOutputPath)" Condition=" '$(PackageReferenceCompatibleProjectStyle)' == 'true' ">
+      <Output TaskParameter="AbsolutePaths" PropertyName="RestoreOutputAbsolutePath" />
+    </ConvertToAbsolutePath>
+    
     <PropertyGroup Condition=" '$(RestoreProjectStyle)' == 'ProjectJson' ">
-      <!-- For project.json projects, only the assets file cache goes in the RestoreOutputPath.  The assets file and the
-           generated .props and .targets go in the project folder.  So in that case use the BaseIntermediateOutputPath,
-           so that the assets file cache will respect a BaseIntermediateOutputPath set in the project body. -->
-      <RestoreOutputPath Condition=" '$(RestoreOutputPath)' == '' " >$(BaseIntermediateOutputPath)</RestoreOutputPath>
+      <!-- For project.json projects, the restore output path is the project directory.  The assets file cache
+           goes in BaseIntermediateOutputPath. -->
+      <RestoreAssetsCacheFolder Condition=" '$(RestoreOutputPath)' == '' " >$(BaseIntermediateOutputPath)</RestoreAssetsCacheFolder>
```

Comment:
Why change this?
I fear this is becoming a bigger refactoring than it needs to be. 
I'd like the change to be more targeted so we can actually merge it, and solve the initial problem.

Reply:
@nkolev92 I just keep hitting issues where I'm having trouble following how all the data about the paths to use flows through, so I try to refactor it to make it more understandable to have more confidence that the change is correct.

Additionally I decided to use the BaseIntermediateOutputPath for project.json projects.  I'm fine with just using MSBuildProjectExtensionsPath in both cases, but I don't think that would significantly affect the other changes.

Reply:
I don't think the refactoring is necesarry, because in the end, this is a very simple change on our side. 

Can you please revert these refactoring and I'll merge it in a feature branch and clean it up there?